### PR TITLE
[GraphBolt] Modify `labels` dtype.

### PR DESCRIPTION
**Pull request by yxy235**
## Description

Modify `labels` dtype from `bool` to default dtype for further use in training.
Update code as recommended in #7068 .

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Comment by mfbalin** [comment]
Why is this dtype change necessary?

**Review by frozenbugs — APPROVED** [review]

File: python/dgl/graphbolt/impl/uniform_negative_sampler.py [hunk 1]
```diff
@@ -86,20 +86,9 @@ def _sample_with_etype(self, node_pairs, etype=None, use_seeds=False):
...
-                        device=seeds.device,
-                    ),
-                    torch.zeros(
-                        neg_num,
-                        dtype=torch.bool,
-                        device=seeds.device,
-                    ),
-                ),
-            )
+            labels = torch.empty(pos_num + neg_num, device=seeds.device)
```
Comment (mfbalin) [inline python/dgl/graphbolt/impl/uniform_negative_sampler.py:89]: Why is using the default dtype better?

**Review by mfbalin — APPROVED** [review]
LGTM.

File: python/dgl/graphbolt/impl/uniform_negative_sampler.py (hunk 1 above)
Comment (yxy235) [reply python/dgl/graphbolt/impl/uniform_negative_sampler.py:89]: When using bool as dtype will cause error  as followed during training  https://github.com/dmlc/dgl/blob/master/examples/sampling/graphbolt/link_prediction.py#L328:
```
Traceback (most recent call last):
  File "/home/ubuntu/dgl/examples/sampling/graphbolt/link_prediction_seeds.py", line 440, in <module>
    main(args)
  File "/home/ubuntu/dgl/examples/sampling/graphbolt/link_prediction_seeds.py", line 422, in main
    train(args, model, graph, features, train_set)
  File "/home/ubuntu/dgl/examples/sampling/graphbolt/link_prediction_seeds.py", line 337, in train
    loss = F.binary_cross_entropy_with_logits(logits, labels)
  File "/home/ubuntu/miniconda3/envs/dgl-dev-gpu-118/lib/python3.10/site-packages/torch/nn/functional.py", line 3165, in binary_cross_entropy_with_logits
    return torch.binary_cross_entropy_with_logits(input, target, weight, pos_weight, reduction_enum)
RuntimeError: Subtraction, the `-` operator, with two bool tensors is not supported. Use the `^` or `logical_xor()` operator instead.
```
So, we should use other dtype instead. I searched previous code and found labels was using default dtype in https://github.com/dmlc/dgl/blob/master/python/dgl/graphbolt/minibatch.py#L482-L484

Comment (mfbalin) [reply python/dgl/graphbolt/impl/uniform_negative_sampler.py:89]: Thank you for the pointers

**Pull request merged**